public/app/core/components/AppChrome/AppChrome.tsx

```diff
@@ -132,6 +135,8 @@ export function AppChrome({ children }: Props) {
       {shouldShowReturnToPrevious && state.returnToPrevious && (
         <ReturnToPrevious href={state.returnToPrevious.href} title={state.returnToPrevious.title} />
       )}
+      <QueryLibraryDrawer />
```

Comment:
this might be a stupid idea, so just putting it out there for discussion 😅 could/should we put these components in the `QueryLibraryDrawerContextProvider` component?
- pros: query library stuff contained in 1 place, probably wouldn't have to share as much state via context (e.g. `drawerOpened`, `activeQuery`, `activeDatasources`, `addQueryModalOpened`)
- cons: maybe couples things together a little too tightly 🤔

Reply:
actually I think it's a pretty good idea. If the drawer/modal components needed to be positioned somewhat specifically it would be a problem but here them being modals this works and makes this much more cohesive.